pkg/executor/aggregate/agg_hash_partial_worker.go

```diff
@@ -31,23 +35,43 @@ import (
 // the number of the worker can be set by `tidb_hashagg_partial_concurrency`.
 type HashAggPartialWorker struct {
 	baseHashAggWorker
+	idForTest int
+	ctx       sessionctx.Context
 
 	inputCh        chan *chunk.Chunk
 	outputChs      []chan *aggfuncs.AggPartialResultMapper
 	globalOutputCh chan *AfFinalResult
-	giveBackCh     chan<- *HashAggInput
-	BInMaps        []int
 
+	// Partial worker transmit the HashAggInput by this channel,
+	// so that the data fetcher could get the partial worker's HashAggInput
+	giveBackCh chan<- *HashAggInput
+
+	BInMaps               []int
 	partialResultsBuffer  [][]aggfuncs.PartialResult
 	partialResultNumInRow int
 
 	// Length of this map is equal to the number of final workers
-	partialResultsMap []aggfuncs.AggPartialResultMapper
-	groupByItems      []expression.Expression
-	groupKey          [][]byte
+	// All data in one AggPartialResultMapper are specifically sent to a target final worker.
+	// e.g. all data in partialResultsMap[3] should be sent to final worker 3.
+	partialResultsMap    []aggfuncs.AggPartialResultMapper
+	partialResultsMapMem atomic.Int64
+
+	groupByItems []expression.Expression
+	groupKey     [][]byte
 	// chk stores the input data from child,
 	// and is reused by childExec and partial worker.
 	chk *chunk.Chunk
+
+	isSpillPrepared       bool
+	spillHelper           *parallelHashAggSpillHelper
+	tmpChksForSpill       []*chunk.Chunk
+	spillSerializeHelpers *aggfuncs.SerializeHelper
+	getNewSpillChunkFunc  func() *chunk.Chunk
+	spillChunkFieldTypes  []*types.FieldType
```

Comment:
Can we put isSpillPrepared, tmpChksForSpill, getNewSpillChunkFunc, and spilChunkFiedlType spillChunkIO into spillHelper?

Reply:
okk

Reply:
> Can we put isSpillPrepared, tmpChksForSpill, getNewSpillChunkFunc, and spilChunkFiedlType spillChunkIO into spillHelper?

okk, but I think we only need to put getNewSpillChunkFunc and spilChunkFiedlType into spillHelper as they can be shared by all workers.